tests/Unit/Domain/CoordinateMaps/Test_UniformCylindricalSide.cpp

```diff
@@ -0,0 +1,517 @@
+                  (0.98 - z_upper_separation) * radius_two) /
+                     (z_plane_frac_plus_one + 1.0),
+                 (z_plane_minus_two - center_two[2] -
+                  (0.98 - z_lower_separation) * radius_two) /
+                     (z_plane_frac_minus_one - 1.0));
+    // max_radius_one_for_alpha_minus is the largest that radius_one can be
+    // and still satisfy alpha_minus > min_alpha_minus.  For
+    // tan(min_alpha_minus) > 0, if max_radius_one_to_fit_inside_sphere_two is
+    // satisfied, then alpha_minus > min_alpha_minus imposes no additional
+    // restriction on radius.
+    const double max_radius_one_for_alpha_plus =
+        min_alpha_plus > 0.5 * M_PI
+            ? std::numeric_limits<double>::max()
+            : std::min(
+                  radius_two * sin(theta_max_plus_two) /
+                      sin(theta_max_plus_one),
+                  (0.98 * radius_two - z_plane_plus_two + center_two[2] -
+                   radius_two * sin(theta_max_plus_two) * tan(min_alpha_plus)) /
+                      (1.0 - cos(theta_max_plus_one) -
+                       sin(theta_max_plus_one) * tan(min_alpha_plus)));
+    const double max_radius_one_for_alpha_minus =
+        min_alpha_minus > 0.5 * M_PI
+            ? std::numeric_limits<double>::max()
+            : std::min(radius_two * sin(theta_max_minus_two) /
+                           sin(theta_max_minus_one),
+                       (0.98 * radius_two + z_plane_minus_two - center_two[2] -
+                        radius_two * sin(theta_max_minus_two) *
+                            tan(min_alpha_minus)) /
+                           (1.0 - cos(theta_max_minus_one) -
+                            sin(theta_max_minus_one) * tan(min_alpha_minus)));
+    CHECK(max_radius_one_for_alpha_minus > 0.0);
+    CHECK(max_radius_one_for_alpha_plus > 0.0);
+    // max_radius_one_to_fit_between_plane_twos is the maximum radius_one
+    // that satisfies the two conditions
+    // z_plane_plus_two >= z_plane_plus_one + z_upper_separation*radius_two
+    // z_plane_minus_two <= z_plane_minus_one - z_lower_separation*radius_two
+    //
+    // This condition is derived from noting that
+    // z_plane_plus_one = center_one[2]+radius_one*z_plane_frac_plus_one
+    // and z_plane_minus_one = center_one[2]+radius_one*z_plane_frac_minus_one
+    // (recall z_plane_frac_minus_one is negative)
+    // and noting that the max value of center_one[2] is >= the min value
+    // of center_one[2].
+    const double max_radius_one_to_fit_between_plane_twos =
+        (z_plane_plus_two - z_plane_minus_two -
+         (z_upper_separation + z_lower_separation) * radius_two) /
+        (z_plane_frac_plus_one - z_plane_frac_minus_one);
+    CHECK(max_radius_one_to_fit_between_plane_twos > 0.0);
+
+    // We add an additional safety factor of 0.99 to
+    // max_radius_one_to_fit_inside_sphere_two so that radius_one
+    // doesn't get quite that large.
```

Comment:
If you don't add this safety factor, will the map fail?

Reply:
The safety factor turns out to be unnecessary.  I removed it and ran the test 10,000 times and got no failures.